extensions/positron-python/python_files/positron/positron_ipykernel/ui.py

```diff
@@ -181,13 +184,31 @@ def open(self, url, new=0, autoraise=True):
 
         # If url is pointing to an HTML file, route to the ShowHtmlFile comm
         if is_local_html_file(url):
+            is_plot = False
+
+            # Send bokeh plots to the plots pane.
+            # Identify bokeh plots by checking the stack for the bokeh.io.showing.show function.
+            # This is not great but currently the only information we have.
+            bokeh_io_showing = sys.modules.get("bokeh.io.showing")
+            if bokeh_io_showing:
+                for frame_info in inspect.stack():
+                    if (
+                        inspect.getmodule(frame_info.frame, frame_info.filename) == bokeh_io_showing
+                        and frame_info.function == "show"
+                    ):
+                        is_plot = True
+                        break
```

Comment:
It's premature at this point but I wonder if it makes sense to have a registry of things like this we can add to in the future that takes the current stack and other appropriate inputs. I can't imagine this is the last time we'll be touching this type of thing.

Reply:
Yeah we may end up with something like that. I'm hoping there are more direct and robust ways to integrate with other libraries, but we'll see!